src/ratbagd-device.c

```diff
@@ -284,40 +284,49 @@ ratbagd_device_get_capabilities(sd_bus *bus,
 	if (r < 0)
 		return r;
 
+#define CHECK_CAP(_cap, _lib_device, _r, _reply)                             \
+	do {                                                                 \
+		if (ratbag_device_has_capability((_lib_device), (_cap))) {   \
+			(_r) = sd_bus_message_append((_reply), "u", (_cap)); \
+			if ((_r) < 0)                                        \
+				return (_r);                                 \
+		}                                                            \
+	} while (0)
+
```

Comment:
not a huge fan of this approach, I don't like macros that return unless it's bleeding obvious that they do.

How about having a 
const enum capability all_caps = [ RDC_QUERY_CONFIGURATION, ..., ]; and a for loop that runs over all those and runs the four lines. Should be possible, right?